### DISCUSSION TITLE
How to generate changelog from the beginning of the project history?

### DISCUSSION THREAD

**Discussion by piotrminkina** [mooltiverse/nyx#296] [Q&A]
Hi,

My project follows Semantic Versioning and Conventional Commit. Several releases the project has already passed. I would like to use nyx, but I have no idea how to generate a Changelog from the beginning of the project history. Each release has a corresponding git tag in the repository. The `./gradlew nyxMake` command generates a Changelog only for the latest version.

Regards
Piotr Minkina

**Comment by flelli** [comment]
Hi @piotrminkina ,

gerenating the entire changelog for all commits in the history is not yet implemented. What is supported is adding new releases to an existing changelog (using the [append](https://mooltiverse.github.io/nyx/guide/user/configuration-reference/changelog/#append) option).

**Reply by piotrminkina** [reply]
@flelli Thanks for reply. I am aware of the option pointed out by you. I just lacked the ability to generate a full changelog. Is it planned to add such feature?

**Comment by flelli** [comment]
Not soon as I'm overwhelmed by my regular job these days but maybe in the future.
Meanwhile there are lots of tools doing that around, and some of them are very popular so I'm pretty confident you can easily get this done

**Reply by piotrminkina** [reply]
OK, thanks for the reply. All the best :)
